pkg/operation/botanist/clusteridentity/clusteridentity.go

```diff
@@ -0,0 +1,124 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusteridentity
+
+import (
+	"context"
+	"fmt"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type shootClusterIdentity struct {
+	clusterIdentity *string
+	seedNamespace   string
+	namespace       string
+	name            string
+	uid             string
+
+	seedClient   client.Client
+	gardenClient client.Client
+
+	logger *logrus.Entry
+}
+
+func New(
+	clusterIdentity *string,
+	name, namespace, seedNamespace, uid string,
+	seedClient, gardenClient client.Client,
+	logger *logrus.Entry,
+) component.Deployer {
+	return &shootClusterIdentity{
+		clusterIdentity: clusterIdentity,
+		name:            name,
+		namespace:       namespace,
+		uid:             uid,
+		seedNamespace:   seedNamespace,
+		seedClient:      seedClient,
+		gardenClient:    gardenClient,
+		logger:          logger,
+	}
+}
+
+func (c *shootClusterIdentity) Deploy(ctx context.Context) error {
+	if c.clusterIdentity == nil {
+		clusterIdentity, err := c.determineShootClusterIdentity(ctx)
+		if err != nil {
+			return err
+		}
+
+		shoot := &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      c.name,
+				Namespace: c.namespace,
+			},
+		}
+
+		c.clusterIdentity = &clusterIdentity
+		patch := []byte(fmt.Sprintf(`{"status":{"clusterIdentity": "%s"}`, clusterIdentity))
+		if err := c.gardenClient.Status().Patch(ctx, shoot, client.RawPatch(types.StrategicMergePatchType, patch)); err != nil {
+			return err
+		}
+	}
+
+	shootClusterIdentity := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      v1beta1constants.ClusterIdentity,
+			Namespace: c.seedNamespace,
+		},
+	}
+
+	_, err := controllerutil.CreateOrUpdate(ctx, c.seedClient, shootClusterIdentity, func() error {
```

Comment:
Why do we need this configmap in the seed? Isn't the `.status.clusterIdentity` in the `Shoot` enough?

Reply:
To be honest I am not sure, when we spoke I was left with the impression we want to have a copy, but I am not certain.

Reply:
@mandelsoft @MartinWeindel did this "request" come from your side?

Reply:
I think the configmap is in the `kube-system` namespace of every cluster in order to be accessible for other components.

Reply:
I removed this, if the colleagues actually need it, I will add it again.

Reply:
Hm, so the cluster identity configmap does only exist in kube-system in garden and shoot, but not in seed?
If this is true then I think we should also deploy it for the seed, to be consistent across our cluster topologys

Reply:
No, I am talking about the copy of the `Shoot` cluster identity in the `Seed` ( in the Shoot's namespace)

Reply:
OK, thanks, then it should be all good! :)